### PR TITLE
LPS-180845 Product QA | Functional Automation - LPS-156495 - As a developer I can operate a system object entry and it's related custom objects in a single request

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-180845

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
the errors are related to a bug which is fixed by https://github.com/liferay-headless/liferay-portal/pull/1170

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 1]
```diff
@@ -203,6 +169,65 @@ definition {
 		return ${response};
 	}
```
Comment (Hijky) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:171]: Can you take a look at it? I'm not sure, Is it the correct macro name? 'nObjectEntryBodyAsNestedField'

**Comment by Hijky** [comment]
For the failures, I checked Jenkins results, I found there were no student entries created by post, only a get request for the student.

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 2]
```diff
@@ -203,6 +169,65 @@ definition {
...
+	macro nObjectEntryBodyAsNestedField {
+		Variables.assertDefined(parameterList = "${body},${nestedField},${numberOfRelatedObjectEntries},${relatedEntryFieldName}");
+
+		var size = ${numberOfRelatedObjectEntries};
+
+		if (!(isSet(externalReferenceCode))) {
+			var externalReferenceCode = "";
+		}
+
+		if (isSet(oneToManyChild) && (${oneToManyChild} == "true")) {
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:181]: The only thing related to the codes for me is I think here set ${oneToManyChild} == "true" in the condition seems enough?

**Comment by yanan** [comment]
> For the failures, I checked Jenkins results, I found there were no student entries created by post, only a get request for the student.

Yes, seems there are 3 realted failed tests on CI. @magjed4289

**Comment by magjed4289** [comment]
> > For the failures, I checked Jenkins results, I found there were no student entries created by post, only a get request for the student.
> 
> Seems there are 1 related failed tests on CI expect for known issues on patch? @magjed4289
> 
> * CanUpdateUserAndCreateRelatedStudentWithPutChildSystemObjectInOneToManyRelationship

the error is related to the issue that will be fixed in the errors are related to a bug which is fixed by https://github.com/liferay-headless/liferay-portal/pull/1170

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro (hunk 2 above)
Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:181]: you mean to get rid of (isSet(oneToManyChild)?

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro (hunk 1 above)
Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:171]: yes, it is correct, the 'n' stands for an undefined number here.

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro (hunk 2 above)
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:181]: yes :small_airplane:

Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:181]: yeah, I though about it is just looked so complete with the other one 😆 Will remove it now cause yeah it is unnecessary :D

**Review by Hijky — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**